analyzer/consensus/consensus.go

```diff
@@ -380,8 +422,32 @@ func (m *processor) queueBlockInserts(batch *storage.QueryBatch, data *consensus
 		data.Epoch,
 		data.GasLimit,
 		data.SizeLimit,
+		proposerAddr,
 	)
 
+	if cmtMeta.LastCommit != nil && cmtMeta.LastCommit.BlockID.IsComplete() {
+		prevSigners := make([]string, 0, len(cmtMeta.LastCommit.Signatures))
+		for _, cs := range cmtMeta.LastCommit.Signatures {
+			if cs.Absent() {
+				continue
+			}
+			entity, ok := consensusToEntity[cs.ValidatorAddress.String()]
+			if !ok {
+				m.logger.Warn("could not convert block signer address to entity id (address not found)",
+					"height", data.BlockHeader.Height,
+					"signer", cs.ValidatorAddress.String(),
+				)
+				continue
+			}
+			prevSigners = append(prevSigners, entity.String())
+		}
+		batch.Queue(
+			queries.ConsensusBlockAddSigners,
```

Comment:
This doesn't work correctly in the fast-sync case. When in fast-sync, blocks are not processed in order, so the previous block (`cmtMeta.LastCommit.Height`) might not exist, and therefore this query won't insert any singers.

Some quick suggestion on how to fix this:
- doing an Upsert, and making all other fields NULL-able (I think a previous version of this PR was done in that way)
- storing the signers in a separate table 

I noticed this in E2E tests of https://github.com/oasisprotocol/nexus/pull/764 since some blocks are missing signers in the `edenfast` test case.